src/geometric_utils.ts

```diff
@@ -71,38 +98,99 @@ export class GeometricUtils {
         }
 
         // Distance from point to line
-        var dst = Math.abs(a*ref_x + b*ref_y + c)/Math.sqrt(a*a + b*b);
+        let dst: number = Math.abs(a*ref_x + b*ref_y + c)/Math.sqrt(a*a + b*b);
         
         // Proportion of the length of segment from p1 to the nearest point
-        const seg_length = Math.sqrt((kp2[0] - kp1[0])*(kp2[0] - kp1[0]) + (kp2[1] - kp1[1])*(kp2[1] - kp1[1]));
-        const kprop = Math.sqrt((nrx - kp1[0])*(nrx - kp1[0]) + (nry - kp1[1])*(nry - kp1[1]))/seg_length;
+        const seg_length: number = Math.sqrt((kp2[0] - kp1[0])*(kp2[0] - kp1[0]) + (kp2[1] - kp1[1])*(kp2[1] - kp1[1]));
+        const kprop: number = Math.sqrt((nrx - kp1[0])*(nrx - kp1[0]) + (nry - kp1[1])*(nry - kp1[1]))/seg_length;
 
-        // return object with info about the point
+        // Return object with info about the point
         return {
             "dst": dst,
             "prop": kprop
         };
     }
 
+    // Check if two line segments intersect
+    public static line_segments_intersect(line1: LineSegment2D, line2: LineSegment2D): boolean {
```

Comment:
You ever consider using a library for this type of thing? I think Turf `lineIntersect` would fit the bill nicely:
https://turfjs.org/docs/#lineIntersect

Reply:
I guess Turf is probably explicitly geospatial, which is not desirable here, so perhaps it's not a great fit after all. I guess the question still stands though — did you investigate whether a library fits the bill for this type of geometry utility function?

Reply:
A couple candidates, for example:
https://github.com/HarryStevens/geometric
https://github.com/mathigon/euclid.js

Reply:
A lot of these functions were first drafts and are not actually used anywhere. Removed in [ef7fda8](https://github.com/SenteraLLC/ulabel/commit/ef7fda8847ae8f262b2915dd0127f3066b0111dd)